Assets/Scripts/GunStats.cs

```diff
@@ -0,0 +1,93 @@
+using UnityEngine;
+
+[CreateAssetMenu(fileName = "GunStats", menuName = "ScriptableObjects/GunStats", order = 1)]
+public class GunStats: ScriptableObject
+{
+
+    // -- IMPORTANT INFO --
+
+    // It is fine to add new properties to this file, but be carefull when REMOVING them
+    // If you remove, for example, magazineSize from these properties and then save, all object data of magazine size 
+    // will be removed in every instance of the sriptable object
+    public enum FireModes
+    {
+        semiAuto,
+        burst,
+        fullAuto
+    }
+    // Projectiles per second
+    [SerializeField]
+    private ModifyableFloat firerate = new ModifyableFloat(5f);
+    public ModifyableFloat Firerate { get => firerate;}
+
+    // Time to reload gun
+    [SerializeField]
+    private ModifyableFloat reloadTime = new ModifyableFloat(3f);
```

Comment:
Actually, I've seen the naming convention for properties be UpperCamelCase. Examples of this can be found all over in the standard packages. I think Unity is the one deviating here...

![image](https://user-images.githubusercontent.com/2846061/201403530-45ed45f5-1a66-4562-b248-45a547d40af2.png)
![image](https://user-images.githubusercontent.com/2846061/201403766-4a1edec2-abf7-438f-bd24-b0185d39c1fe.png)

Reply:
Interesting 🤔 We should probably decide on which to follow in that case

Reply:
Hm, this spices things up :upside_down_face: 
I assumed Unity's naming convention was in line with C#'s and didn't think of the properties that happen to be PascalCase (though they've thrown me for a loop when I've assumed that e.g. array.Count is a method)